src/Backend/SgateShopgatePlugin/Controllers/Frontend/shopgate.php

```diff
@@ -500,6 +500,56 @@ public function checkoutAction()
         }
     }
 
+    /**
+     * Custom action to login user and redirect to bonus points
+     */
+    public function bonusPointsAction()
+    {
+        $sessionId = $this->Request()->getParam('sessionId');
+
+        if (isset($sessionId)) {
+            session_write_close();
+            session_id($sessionId);
+            session_start(array(
+                'sessionId' => $sessionId
+            ));
+        }
+
+        $token = $this->Request()->getParam('token');
+        $this->session->offsetSet('sgWebView', true);
+
+        if ($this->webCheckoutHelper->loginAppUser($token, $this->Request())) {
+            $this->redirect('BonusSystem/points');
```

Comment:
Only this line differs in every method. Can you please create a general one an parametrize this string for every action in this class? I know this should have been done earlier, but now things adding up =)

Reply:
i moved some parts out to a new function